### DISCUSSION TITLE
Managed Identity for TriggerAuthentication when using azureKeyVault?

### DISCUSSION THREAD

**Discussion by AMiron-Bunnings** [kedacore/keda#3039] [Q&A / Need Help] [answered]
Looking at the documentation:

https://keda.sh/docs/2.7/concepts/authentication/#re-use-credentials-and-delegate-auth-with-triggerauthentication

One would think that using the managed identity for azureKeyVault is supported; the comment next to credentials for azureKeyVault reads: _# Optional. Required when not using pod identity._

However, when trying to create a TriggerAuthentication:

```apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: my-trigger-auth
  namespace: my-namespace    
spec:
  podIdentity:
    provider: azure                                     # Optional.         
  azureKeyVault:                                        # Optional.
    vaultUri: https://my-vault-uri/                       
    secrets:                                            # Required.
    - parameter: connectionString
      name: my-secret-name   
```

I always get:

`error validating data: ValidationError(TriggerAuthentication.spec.azureKeyVault): missing required field "credentials" in sh.keda.v1alpha1.TriggerAuthentication.spec.azureKeyVault; if you choose to ignore these errors, turn validation off with --validate=false`

I'm using Keda 2.7.0 and I have a managed identity set-up with the cluster. I have successfully used managed identities with Keda and services like Azure Service Bus.

Is the managed identity for azureKeyVault in this case already implemented? If so, what am I doing wrong?

**Comment by tomkerkhove** [comment, marked as answer]
@v-shenoy DO we need to improve our docs? I presume so given it needs to tell how to auth

**Reply by v-shenoy** [reply]
Yup, this is just me being an idiot. Pod identity support for key vault was added with workload identity which got pushed beyond v2.7.0, and while I reverted the doc changes, I clearly missed some lines.

**Reply by v-shenoy** [reply]
@AMiron-Bunnings It will be in v2.8.0 :) Sorry for the error in docs.

**Reply by AMiron-Bunnings** [reply]
That's great @v-shenoy, thanks for the info!
